File: src/main/java/br/unifor/costify/application/dto/RegisterRecipeCommand.java
package br.unifor.costify.application.dto;

import br.unifor.costify.domain.valueobject.RecipeIngredient;

import java.util.List;

public record RegisterRecipeCommand(
    String name,
    List<RecipeIngredient> ingredients
) {
    
    public RegisterRecipeCommand {
        if (name == null || name.isBlank()) {
            throw new IllegalArgumentException("Recipe name cannot be null or empty");
        }
        if (ingredients == null || ingredients.isEmpty()) {
            throw new IllegalArgumentException("Recipe must have at least one ingredient");
        }
    }
}

File: internal/application/dto/command/register_recipe_command.go
package command

import (
	"fmt"
	"strings"

	"github.com/vini/costify-go/internal/domain/valueobject"
)

// RegisterRecipeCommand represents the command to register a new recipe
type RegisterRecipeCommand struct {
	Name        string                           `json:"name" validate:"required"`
	Ingredients []valueobject.RecipeIngredient   `json:"ingredients" validate:"required,min=1"`
}

// NewRegisterRecipeCommand creates a new RegisterRecipeCommand with validation
func NewRegisterRecipeCommand(
	name string,
	ingredients []valueobject.RecipeIngredient,
) (RegisterRecipeCommand, error) {
	if strings.TrimSpace(name) == "" {
		return RegisterRecipeCommand{}, fmt.Errorf("recipe name cannot be null or empty")
	}
	if len(ingredients) == 0 {
		return RegisterRecipeCommand{}, fmt.Errorf("recipe must have at least one ingredient")
	}

	return RegisterRecipeCommand{
		Name:        name,
		Ingredients: ingredients,
	}, nil
}